hls-plugin-api/src/Ide/Types.hs

```diff
@@ -890,9 +893,52 @@ instance GEq IdeNotification where
 instance GCompare IdeNotification where
   gcompare (IdeNotification a) (IdeNotification b) = gcompare a b
 
+-- | Restricted version of 'LspM' specific to plugins
```

Comment:
Worth explaining why this is here! Something like:

We use this instead of `LspM`, since there are parts of the LSP server state which
plugins should not access directly, but instead only via the build system.

Reply:
I've added your explanation to the comment!